src/plugins/PluginsContext.ts

```diff
@@ -0,0 +1,24 @@
+import {createContext} from 'react';
+
+export interface PluginData {
+  getComponent: (props?: any) => any;
```

Comment:
Why the `PluginData` would have `getComponent`? `PluginData` should rather never be a component (more likely React element), and sometimes should be arrays or array items (i.e. tabs definition).

Reply:
We need a way to pass props to the injected components from the parent component (e.g. test execution id, status, etc). As we don't want to propagate them to the top of the component tree, I've added this render function that can be called within the parent component with the needed props and it will return the component with the needed state

Reply:
It could be easily handled with Context providers/consumers.

Reply:
By decentralized, I mean that in *core* you should give flexibility for the plugins. The *core* shouldn't care if the plugin will display some data as a modal, a tab, or in different window - *core* only has to provide essential data for the display.

Keeping it like `props` would mean that the plugin will have this data only in a single place, so it won't be simple to handle it throughout the plugin as a whole.